flag_engine/features/schemas.py

```diff
@@ -63,3 +64,20 @@ def make_feature_state(self, data, **kwargs) -> FeatureStateModel:
         feature_state = FeatureStateModel(**data)
         feature_state.set_value(value)
         return feature_state
+
+    @post_dump()
+    def validate_percentage_allocations(self, data, **kwargs):
+        """Since we do support modifying percentage allocation on a per identity override bases
+        we need to validate the percentage before building the document(dict)"""
+        if (
```

Comment:
I thikn the following would be slightly neater / more readable. 

```python
total_allocation = sum(
    mvfsv["percentage_allocation"] for mvfsv in data["multivariate_feature_state_values"]
)
if total_allocation > 100:
    raise InvalidPercentageAllocation(...)
return data
```

I'm also worried that this would currently throw a HTTP 500 which it would be good to avoid where possible. Do we know which places this could possibly be thrown, can we wrap it with a try catch at the handler level perhaps?

Reply:
Yes, it will be handled by Django. here is a snipped of the code using this

```
        try:
            identity_dict = build_identity_dict(identity)
        except InvalidPercentageAllocation as e:
            raise serializers.ValidationError(
                {
                    "multivariate_feature_state_values": "Total percentage allocation"
                    "for feature must be less than 100 percent"
                }
            ) from e
        Identity.dynamo_wrapper.put_item(identity_dict)

```

Reply:
Ok, but that code doesn't exist yet I guess? We just need to remember to include it.

Reply:
No, that does exist in WIP PR here: https://github.com/Flagsmith/flagsmith/pull/584/files